### ISSUE TITLE
Add customizable health/healing options

### ISSUE THREAD

**Issue by rdroog** [DrHenchman/uhc-pack#71]
**Is your feature request related to a problem? Please describe.**
It's quite hard to live with the current amount of hearts for new players, and the easy difficulty in Minecraft is a bit too easy (no poisonous spiders). To solve this having customizable health/healing options would be great. Other UHC packs also have some options for this

**Describe the solution you'd like**
* Set maximum heath to a different amount. Both higher and lower (maybe in the range of 50%-300% of current health).
* Set healing health of golden appels/other items to a different amount. Same as above; 50%-300% would be great.

**Describe alternatives you've considered**
* Using another UHC pack: the ones we've tried don't allow for setting a shrinking border start in the future. We like that option of this pack too much to use another pack.
* Setting difficulty to easy or medium: easy is just too easy, medium is okay but still inexperienced people die too soon in our opinion.

**Additional context**
We love this datapack so I'd be great if this feature would be added. If you have any pointers I can maybe try to create PR for it, but I have no experience in datapacks (I do have programming experience).

**Comment by DrHenchman** [comment]
Great to hear you have found the pack useful. This sounds like a reasonable feature to add.

Would you expect that this would work as a handicap (newer players start with more health, whereas experience players don't) or would all players spawn with additional health?

**Comment by DrHenchman** [comment]
How this would be implemented.

The option would need to be added similar to other variable configurations with both an enabled/disabled toggle and a number of additional hearts. It seems that the `minecraft:health_boost` effect only allows for you to add hearts in 2 heart increments, so possible you could add 2, 4, 6, 8 or 10 additional hearts. Reducing the number of hearts would be difficult (as it would involve damaging the player the right number of damage to reduce them below 10 hearts). There are existing examples of such as night_vision and conduit_power of how to give effects to players based on a configuration value.

Here is an example of a recent change I made to add conduit power: https://github.com/DrHenchman/uhc-pack/commit/26e40450aaaf63b0d6df000d91472f17b690b9ce. This is only a binary toggle though. You would need to look at other configuration examples for how to add it as a sliding scale (the shrinking border configuration is a good example for that)

You can find the contributor guide here: https://github.com/DrHenchman/uhc-pack/blob/master/CONTRIBUTING.md. Otherwise I regularly go through phases of implementing feature requests when I have spare time, so if you'd prefer to wait this is fine as well :)

**Comment by rdroog** [comment]
> Great to hear you have found the pack useful. This sounds like a reasonable feature to add.
> 
> Would you expect that this would work as a handicap (newer players start with more health, whereas experience players don't) or would all players spawn with additional health?

No, we would like it to be equal for everyone, just giving everyone a bit of extra breathing space in terms of life.

**Comment by rdroog** [comment]
@DrHenchman: I've implemented this now. I couldn't find a way to change how much a golden apple (or other healing item) gives, so I've left that alone. And we only need more starting health, and as you said getting to <10 hearts is hard I haven't tried that either.
